module/shared-kernel/src/Domain/Collection/EmailCollection.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\SharedKernel\Domain\Collection;
+
+use Ergonode\SharedKernel\Domain\ValueObject\Email;
+
+/**
+ * @todo Waiting for decision about collection library
+ */
+class EmailCollection
```

Comment:
If we want to have specific generic types I'd go with the suggested library - it's quite a good idea I guess.

Otherwise, I'd just stick with array and appropriate PHPDoc hints i.e. `@var Mail[] $mails` and not dedicated classes - such would just increase clutter with no much of a advantages

Reply:
Yes, i will use Ramsey Collection